plugin/src/main/kotlin/org/gradleweaver/plugins/jlink/JLinkTask.kt

```diff
@@ -4,132 +4,257 @@ import org.gradle.api.DefaultTask
 import org.gradle.api.Project
 import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.RegularFileProperty
-import org.gradle.api.tasks.InputFile
-import org.gradle.api.tasks.OutputDirectory
-import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.*
 import org.gradle.internal.jvm.Jvm
+import org.gradle.internal.os.OperatingSystem
```

Comment:
❌ Internal API usage.

Reply:
Unfortunately, this is a very useful API with no real public alternative. I could copy the class into the plugin, but that seems hacky

Reply:
https://github.com/gradle/gradle/issues/7735